src/librustdoc/passes/collect_intra_doc_links.rs

```diff
@@ -435,8 +435,10 @@ impl<'a, 'tcx> LinkCollector<'a, 'tcx> {
 
         // Try looking for methods and associated items.
         let mut split = path_str.rsplitn(2, "::");
```

Comment:
should we use `rsplit_once` here?

Reply:
I wanted to, but had a hard time getting this code working properly. My WIP ended up being more confusing imo, so I gave up.

Reply:
Yeah everything here is a mess. I tried to clean it up in https://github.com/rust-lang/rust/pull/76467 but it ended up not working and getting stuck for 3 months :( so I reverted it.